### ISSUE TITLE
[Bug] x:Bind build error when used on control from another namespace

### ISSUE THREAD

**Issue by MitchRazga** [levitali/CompiledBindings#9]
I seem to have found an odd bug that occurs when using x:Bind on a control from another namespace where the end of one namespace matches the start of another namespace eg.  CompiledBindingsTest.**_WPF_** and **_WPF_**.CustomControls.

The following error will occur when attempting to build:
`error CS0234: The type or namespace name 'CustomControls' does not exist in the namespace 'CompiledBindingsTest.WPF' (are you missing an assembly reference?)`
It's like part of the namespace is being cut off.

Please see the attached repro:
[CompiledBindingsTest.zip](https://github.com/levitali/CompiledBindings/files/8255230/CompiledBindingsTest.zip)

There are 2 WPF apps in this solution. They are the the exact same except the namespace is different.
CompiledBindingsTest will build successfully but CompiledBindingsTest.WPF will not.
Renaming the namespace to something like CompiledBindingsTest.WPF2 will allow it to build successfully.

Thanks for creating this project, I've been using it for a few days now and it's great!

**Comment by levitali** [comment]
Looks like this is a strange WPF bug.  Try removing all x:Bind from the project. You could also remove completely the CompiledBindings nuget to be sure. And then name the ExternalUserControl. For example:

<customcontrols:ExternalUserControl Margin="20" x:Name="test"/>

You will receive the same error.

CompiledBindings needs controls to be named (names are added automatically during compilation if missing) to reference the controls in the generated C# files.

I think you will have to change you namespaces in order to overcome this WPF bug.

**Comment by MitchRazga** [comment]
Yeah I've done some more research into this and it definitely looks like a WPF bug. There were a few comments on Stack Overflow of other people experiencing this and they all worked around it by renaming the namespace.

Looking into the obj folder for each project I was able to compare all the auto-generated code for the MainWindow across the two apps and they are exactly the same except for the namespace difference.

I tried a few workarounds like setting a XmlnsDefinition in the external namespace assembly and using a schema but no luck.
